### PR TITLE
Shows unassociated roles for Credential

### LINKED ISSUE #8840: [ui_next] No feedback given in Add User Role wizard if permission cannot be applied
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
Giving users access to credentials, it’s really easy to hit this

{"msg":"You cannot grant credential access to a user not in the credentials' organization"}

This is not surfaced properly in the UI for adding roles here:

Screen Shot 2020-12-10 at 5 18 31 PM

I get no indication of the error other than the network console. Yes, there are 4 different roles being applied here
Read to joe
Use to joe
Read to elmo
Use to elmo

So you would need to give some “scorecard” type thing, recognizing the possibility that some permission assignments were successful and some were not.

##### ENVIRONMENT
* AWX version: 12/10

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
Resolves #8840 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
permissions

### CODE REVIEW

File: awx/ui_next/src/components/AddRole/AddResourceRole.jsx [hunk 1]
```diff
@@ -76,22 +77,58 @@ function AddResourceRole({ onSave, onClose, roles, resource, onError }) {
     setCurrentStepId(step.id);
   };
 
+  const userOrTeamAndCredentialShareOrg = async (userOrTeam, role) => {
```
Comment (AlexSCorey) [inline awx/ui_next/src/components/AddRole/AddResourceRole.jsx:130]: would love help on rename of `userOrTeam` as it doesn't roll off the tongue. I thought about `userUnit` but thought that didn't make much sense either.....suggestions?

Comment (jakemcdermott) [reply awx/ui_next/src/components/AddRole/AddResourceRole.jsx:130]: @AlexSCorey just call it something like `sharesCredentialOrg` and let the additional context be described by the variable you call it with.

`sharesCredentialOrg(user)`
`sharesCredentialOrg(team)`  is plenty descriptive

File: awx/ui_next/src/components/AddRole/AddResourceRole.jsx [hunk 2]
```diff
@@ -76,22 +76,59 @@ function AddResourceRole({ onSave, onClose, roles, resource, onError }) {
...
+    if (selectedResource === 'users') {
+      const {
+        data: { results },
+      } = await UsersAPI.readOrganizations(userOrTeam.id);
+
+      if (results.some(r => r.id === resource.organization)) {
+        orgCredCheck = UsersAPI.associateRole(userOrTeam.id, role.id);
+      } else {
+        orgCredCheck = {
+          headerRow: [t`Username`, t`Frst name`, t`Last name`, t`Role`],
```
Comment (nixocio) [inline awx/ui_next/src/components/AddRole/AddResourceRole.jsx:141]:
```suggestion
          headerRow: [t`Username`, t`First name`, t`Last name`, t`Role`],
```

**Review by nixocio — APPROVED** [review]
Small typo. Other than that looks good.

**Review by jakemcdermott — APPROVED** [review]

**Comment by nixocio** [comment]
recheck

**Comment by akus062381** [comment]
@nixocio I am working on a YOLO build right now to test :-)

**Review by akus062381 — APPROVED** [review]
manually tested. looks good!

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
recheck

**Pull request closed without merging**